py/dml/c_backend.py

```diff
@@ -1238,7 +1238,7 @@ def generate_simple_events_control_methods(device):
 
     site = logging.SimpleSite('<_cancel_simple_events>')
     by_dims = {}
-    for hook in dml.globals.hooks:
+    for hook in objects.Device.hooks:
```

Comment:
You've earlier pushed for global variables being exclusively inside `dml.globals` for the potential future development of a DMLC daemon. Have you changed your mind?

Reply:
The idea with moving stuff to `globals` is essentially to eventually collect that to only be accessible through some context manager, because that resolves most of the problems with globality. Now I lock stuff behind a different context manager with a narrower scope.